aerogear/ups-test-suite
# Extends dynamic-test-driver functionality | Adds new UPS Sender Endpoint

### Dynamic-Test-Driver
Implements the functionality to read a CSV file with device tokens and send 1 push notification to each. Also to specify the delay between requests and to instantiate many test runners simultaneously.

All similar to the already implemented [rhmap-test-driver](https://github.com/aerogear/ups-test-suite/tree/master/rhmap-test-driver).

In order to verify it:
1. Start UPS up.
2. Create mocks with [mock-data-loader](https://github.com/aerogear/ups-test-suite/tree/master/mock-data-loader), saving them into a devices.csv file.
3. Run `node app/index.js -a <app id> -m <app master secret> -c ./devices.csv -d 10000 -i 5`
4. Verify that every 10 seconds there are 5 new notifications.

### RHMAP-Test-Driver
The Cloud App now has [a new route](https://github.com/josemigallas/rhmap-test-cloud-app/commit/4b046ae6ad5841fc76d221dca976fc60e3de2071) to invoke the UPS Sender API directly, without using `fh.push()`. The rhmap-test-driver has been upgraded to use this new route by adding the flag `-D`. It is also needed to specify the `pushApplicationID` and the `masterSecret` of the mobile application, that can be found under *push -> Sender API* in Studio.

### Logs
Logs have been improved in general. An ERROR should be an exception triggered by either the CloudApp or the TestDriver. Any other result will be considered a RESPONSE, regardless of it was successful or not.

### :warning: Yet More Changes
Added implementation to cover https://issues.jboss.org/browse/RHMAP-13659. Now by adding `-t` flag, all notifications are set via `sender.sendBatch()` method.

Also the command line parser has been extracted into a utility method.
